front_end_pocket_poll/lib/groups_widgets/group_categories.dart

```diff
@@ -338,71 +318,78 @@ class _GroupCategoriesState extends State<GroupCategories> {
 
   // adds this category to the list of selected (i.e. adds it to the group)
   void selectCategory(Category category) {
-    setState(() {
-      if (widget.selectedCategories.keys.contains(category.categoryId)) {
-        widget.selectedCategories.remove(category.categoryId);
-      } else {
-        widget.selectedCategories
-            .putIfAbsent(category.categoryId, () => category.categoryName);
-      }
-    });
+    if (widget.selectedCategories.keys.contains(category.categoryId)) {
+      widget.selectedCategories.remove(category.categoryId);
+    } else {
+      widget.selectedCategories
+          .putIfAbsent(category.categoryId, () => category.categoryName);
+    }
+    this.getCategoriesAndBuildRows();
   }
 
   /*
-    Loads all the categories the group has attached to it. Categories that are owned by the active user
-    are set to checked if they are in the group.
+    Loads all the categories the group has attached to it. Categories that are
+    owned by the active user are set to checked if they are in the group.
    */
-  void getCategories() async {
-    ResultStatus<List<Category>> resultStatus =
-        await CategoriesManager.getAllCategoriesList();
-    this.loading = false;
-    if (resultStatus.success) {
-      this.errorLoading = false;
-      List<Category> selectedCats = resultStatus.data;
+  void getCategoriesAndBuildRows() async {
+    //only query the group categories one time per build of this widget
+    if (this.groupCategories == null) {
+      ResultStatus<List<Category>> resultStatus =
+          await GroupsManager.getAllCategoriesList(
+              Globals.currentGroup.groupId);
+      this.loading = false;
+
+      if (resultStatus.success) {
+        this.groupCategories = resultStatus.data;
+        this.errorLoading = false;
+      } else {
+        setState(() {
+          this.errorWidget = categoriesError(resultStatus.errorMessage);
+          this.errorLoading = true;
+        });
+      }
+    }
+
+    if (!this.errorLoading) {
+      this.ownedCategoryRows.clear();
+      this.groupCategoryRows.clear();
+
       int index = 0; // used for integration tests
-      for (Category category in selectedCats) {
-        if (!Globals.user.ownedCategories.contains(category) &&
-            category.groups.containsKey(Globals.currentGroup.groupId)) {
+
+      //build the users's categories
```

Comment:
So continuing from my previous comment. I'm saying that essentially right here this would be a separate method called something like "buildCategoryRows"